components/server/src/authorization/spicedb.ts

```diff
@@ -40,63 +34,67 @@ export function spiceDBConfigFromEnv(): SpiceDBClientConfig | undefined {
     private client: Client | undefined;
 
-    private readonly interceptors: grpc.Interceptor[] = [];
-
     constructor(
-        private readonly _clientConfig: SpiceDBClientConfig,
-        private readonly clientCallMetrics?: IClientCallMetrics | undefined,
-    ) {
-        if (this.clientCallMetrics) {
-            this.interceptors.push(createClientCallMetricsInterceptor(this.clientCallMetrics));
-        }
-    }
+        private readonly clientConfig: SpiceDBClientConfig,
+        private readonly interceptors: grpc.Interceptor[] = [],
+    ) {}
 
     getClient(): SpiceDBClient {
-        let client = this.client;
-        if (!client) {
-            client = spicedbClientFromConfig(this.clientConfig);
-        } else if (client.getChannel().getConnectivityState(true) !== grpc.connectivityState.READY) {
-            // (gpl): We need this check and explicit re-connect because we observe a ~120s connection timeout without it.
-            // It's not entirely clear where that timeout comes from, but likely from the underlying transport, that is not exposed by grpc/grpc-js
-            client.close();
-
-            log.warn("[spicedb] Lost connection to SpiceDB - reconnecting...");
-
-            client = spicedbClientFromConfig(this.clientConfig);
+        if (this.client) {
+            const state = this.client.getChannel().getConnectivityState(true);
+            if (state === grpc.connectivityState.TRANSIENT_FAILURE || state === grpc.connectivityState.SHUTDOWN) {
+                log.warn("[spicedb] Lost connection to SpiceDB - reconnecting...");
+                try {
+                    this.client.close();
+                } catch (error) {
+                    log.error("[spicedb] Failed to close client", error);
+                } finally {
+                    this.client = undefined;
+                }
+            }
         }
-        this.client = client;
-
-        return client.promises;
-    }
 
-    protected get clientConfig() {
-        const config = this._clientConfig;
-        if (this.interceptors) {
-            return {
-                ...config,
-                options: {
-                    interceptors: [...this.interceptors],
+        if (!this.client) {
+            this.client = v1.NewClient(
+                this.clientConfig.token,
+                this.clientConfig.address,
+                v1.ClientSecurity.INSECURE_PLAINTEXT_CREDENTIALS,
+                undefined, //
+                {
+                    // wie ping frequently to check if the connection is still alive
+                    "grpc.keepalive_time_ms": 1000,
+                    "grpc.keepalive_timeout_ms": 1000,
+
+                    "grpc.max_reconnect_backoff_ms": 5000,
+                    "grpc.initial_reconnect_backoff_ms": 500,
+                    "grpc.service_config": JSON.stringify({
+                        methodConfig: [
+                            {
+                                name: [{}],
```

Comment:
This seems as if it's not applied; even with decreasing this I never see a DEADLINE_EXCEEDED.

The docs say this is a server-side config ([source](https://github.com/grpc/grpc/blob/master/doc/service_config.md#objective)), which is only used as fallback by clients if the server does not provide default values ([source](https://github.com/grpc/grpc/blob/master/doc/service_config.md#apis)).

IMO we should keep the explicit deadline on per-call basis.